File: archive/canvas-api/src/main/java/com/instructure/canvasapi/model/LaunchDefinition.java
/*
 * Copyright (C) 2017 - present Instructure, Inc.
 *
 *     Licensed under the Apache License, Version 2.0 (the "License");
 *     you may not use this file except in compliance with the License.
 *     You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *     Unless required by applicable law or agreed to in writing, software
 *     distributed under the License is distributed on an "AS IS" BASIS,
 *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *     See the License for the specific language governing permissions and
 *     limitations under the License.
 *
 */
package com.instructure.canvasapi.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

public class LaunchDefinition implements Parcelable {

    @SerializedName("definition_type")
    public String definitionType;

    @SerializedName("definition_id")
    public Integer definitionId;

    @SerializedName("name")
    public String name;

    @SerializedName("description")
    public String description;

    @SerializedName("domain")
    public String domain;

    @SerializedName("placements")
    public Placements placements;

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.definitionType);
        dest.writeValue(this.definitionId);
        dest.writeString(this.name);
        dest.writeString(this.description);
        dest.writeString(this.domain);
        dest.writeParcelable(this.placements, flags);
    }

    public LaunchDefinition() {
    }

    protected LaunchDefinition(Parcel in) {
        this.definitionType = in.readString();
        this.definitionId = (Integer) in.readValue(Integer.class.getClassLoader());
        this.name = in.readString();
        this.description = in.readString();
        this.domain = in.readString();
        this.placements = in.readParcelable(Placements.class.getClassLoader());
    }

    public static final Parcelable.Creator<LaunchDefinition> CREATOR = new Parcelable.Creator<LaunchDefinition>() {
        @Override
        public LaunchDefinition createFromParcel(Parcel source) {
            return new LaunchDefinition(source);
        }

        @Override
        public LaunchDefinition[] newArray(int size) {
            return new LaunchDefinition[size];
        }
    };
}

File: libs/canvas-api-2/src/main/java/com/instructure/canvasapi2/models/LaunchDefinition.kt
/*
 * Copyright (C) 2017 - present Instructure, Inc.
 *
 *     Licensed under the Apache License, Version 2.0 (the "License");
 *     you may not use this file except in compliance with the License.
 *     You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *     Unless required by applicable law or agreed to in writing, software
 *     distributed under the License is distributed on an "AS IS" BASIS,
 *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *     See the License for the specific language governing permissions and
 *     limitations under the License.
 *
 */
package com.instructure.canvasapi2.models

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class LaunchDefinition(
        @SerializedName("definition_type")
        var definitionType: String,
        @SerializedName("definition_id")
        var definitionId: Long? = null,
        var name: String?,
        var description: String?,
        var domain: String?,
        var placements: Placements?,
        var url: String?
) : Parcelable {

    companion object {
        const val GAUGE_DOMAIN = "gauge.instructure.com"
        const val STUDIO_DOMAIN = "arc.instructure.com" // NOTE: The subdomain hasn't changed to reflect the rebranding of Arc -> Studio yet
        const val MASTERY_DOMAIN = "app.masteryconnect.com"
        const val PORTFOLIO_DOMAIN = "iad.portfolio.instructure.com"
    }
}
